### PR TITLE
Fix text overflow for "Excerpts of tag pages are cut off"

### LINKED ISSUE #7006: Excerpts of tag pages are cut off
### Please describe the problem (or idea)

> What did you expect to see that you didn't?
I was trying to go on the topic pages of certain tags like this one:

excerpt-bug

I noticed that the excerpts of the wiki descriptions are cut off for several of them as you can see above. I think we should try and fix this so the full paragraph is displayed.

### Please show us where to look

The code that decides this is at this line of the show tag view:
https://github.com/publiclab/plots2/blob/562e17306d0ddafb592bb8ad08a57ed73f8995c1/app/views/tag/show.html.erb#L23.
It takes the body of the wiki description and takes the first paragraph using the ```split()``` function.
I don't know why this isn't working but am going to try and figure it out. I would love your help :smile:.

****

## Thank you!

Your help makes Public Lab better! We *deeply* appreciate your helping refine and improve this site. 

To learn how to write really great issues, which increases the chances they'll be resolved, see:

https://publiclab.org/wiki/developers#Contributing+for+non-coders

### PR DESCRIPTION
**Pull request by sashadev-sky**
Fixes #7300  (<=== Add issue number here)

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

* [ ] PR is descriptively titled 📑 and links the original issue above 🔗
* [ ] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `rake test`
* [ ] code is in uniquely-named feature branch and has no merge conflicts 📁
* [ ] screenshots/GIFs are attached 📎 in case of UI updation
* [ ] ask `@publiclab/reviewers` for help, in a comment below

> We're happy to help you get this ready -- don't be afraid to ask for help, and **don't be discouraged** if your tests fail at first!

If tests do fail, click on the red `X` to learn why by reading the logs.

Please be sure you've reviewed our contribution guidelines at https://publiclab.org/contributing-to-public-lab-software 

Thanks!

### CODE REVIEW

**Comment by sashadev-sky** [comment]
@publiclab/plots2-reviewers @Uzay-G @sharmamansi 
Screen Shot 2020-01-18 at 11 02 11 PM

This PR basically just takes the first 18 words, adds them to their own `span` tags in groups of 6, then adds a "..." onto the end of the last `span`. The text will never go outside the block it will just keep creating new lines. 

I think there should be a follow up issue requesting to section out the editor with a separate section for the tag "label" with a character limit

image

**Comment by Uzay-G** [comment]
I think the string of functions you called on it are sometimes turning it into a nil object. For example, with this error: ERROR["test_tag_show", #<Minitest::Reporters::Suite:0x00005619b310b3c8 @name="TagControllerTest">, 61.01238731300003]

```
 test_tag_show#TagControllerTest (61.01s)

Minitest::UnexpectedError:         ActionView::Template::Error: undefined method `join' for nil:NilClass

            app/views/tag/show.html.erb:24:in `block in _app_views_tag_show_html_erb__3232351324576816542_47334236386020'
            app/views/tag/show.html.erb:23:in `times'
            app/views/tag/show.html.erb:23:in _app_views_tag_show_html_erb__3232351324576816542_47334236386020'
            app/controllers/tag_controller.rb:187:in `block (2 levels) in show'
            app/controllers/tag_controller.rb:186:in `show'
            test/functional/tag_controller_test.rb:180:in `block in <class:TagControllerTest>'
```

File: app/views/tag/show.html.erb [hunk 1]
```diff
@@ -20,7 +20,9 @@
 
       <% end %>
       <% if @wiki %>
-        <p class = "wiki-sub"><%= @wiki.latest.body.split('\n').first %></p>
+        <% 3.times do |i| %>
+          <span class="wiki-sub"><%= @wiki.latest.body.split(' ').slice!((i * 6)..(i * 6) + 5).join(' ') %></span>
```
Comment (Uzay-G) [inline app/views/tag/show.html.erb]: I found a method that actually does this for us: https://apidock.com/rails/ActionView/Helpers/TextHelper/truncate.

It truncates the string and automatically adds `...`. We should use this I think

Comment (sashadev-sky) [reply app/views/tag/show.html.erb]: ok will look into it! Thanks for the review :)

**Comment by jywarren** [comment]
Thanks! We just overhauled this page so I think we are ok now!

**Pull request closed without merging**